.github/workflows/docker.yml

```diff
@@ -21,33 +24,253 @@ jobs:
             echo "has-secrets=1" >> "$GITHUB_OUTPUT"
             echo "has secrets!"
           else
-            echo "has-secrets=0" >> "$GITHUB_OUTPUT"
             echo "no secrets!"
           fi
```

Comment:
This did not work as expected. The check `if: needs.config.outputs.has-secrets` returns true in both cases.

Either check for `if: needs.config.outputs.has-secrets == 1` or don't set `has-secrets` when there are no secrets. I went with the latter.